src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -40,6 +40,9 @@
     /** String literal for text attribute. */
     private static final String TEXT_ATTRIBUTE_NAME = "text";
 
+    /** String literal for text attribute. */
+    private static final String VALUE_ATTRIBUTE_NAME = "value";
```

Comment:
I lost a track of decision on new attribute reason.
Why we need `value` in addition to `text` ? What is the difference ? Why we can not keep content of file in one attribute ?

we might need to put this info some xdoc, users need to understand  how to write xpath expressions.

Reply:
@text - about naming 
@value - about value 
As @lars-sh proposed, we can now use the following queries to get `NUM_INT` token.
```
int abc = 123;

//VARIABLE_DEF[@text='abc']//NUM_INT
//VARIABLE_DEF//NUM_INT[@value=123]
```

Another use case, next example demonstrates how we can query same `IDENT` node
```
//CLASS_DEF[@text='SomeClass']/IDENT
//CLASS_DEF/IDENT[@value='SomeClass']
```

Now the main reason we created this attribute
For example we want to suppress all `import` statements, which contains `java` in the path
```
import java.util.*;
```
Ast tree of this import
```
IMPORT -> import [1:0]
|--DOT -> . [1:16]
|   |--DOT -> . [1:11]
|   |   |--IDENT -> java [1:7]
|   |   `--IDENT -> util [1:12]
|   `--STAR -> * [1:17]
`--SEMI -> ; [1:18]
```
Previous implementation didn't allow it, but with current it would smth like 
`//IMPORT[.//IDENT[@value='java']]`

@lars-sh please, correct me if I am wrong

Reply:
@romani 
I will add examples to documentation, but what about `XpathQueryGenerator` should it be changed it this PR as well?

Reply:
@romani From my point of view the `value` attribute technically fulfills all needs, though keeping the `text` attribute allows to offer backwards compatibility.

Reply:
> I will add examples to documentation,

please do, now `value` looks reasonable and clear to use based on tree structure and `text` blow my mind.
Please update documentation to make it clear what attribute to use in what situation.

Reply:
done
@romani @rnveach 
I updated documentation and `XpathQueryGenerator`